### PR TITLE
Add support for Scala 2.11/2.13 in worksheets

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, worksheets only worked for Scala 2.12.10. Now, worksheets
support all Scala versions where completions are supported.

Another nice improvements in this change is that we no longer poke into
internal mdoc APIs. The worksheet evaluation logic has been added to
public mdoc APIs so that we have a clear separation of concerns. This
means that improvements to rendering of values in worksheets need to be
contributed to the scalameta/mdoc repository instead of
scalameta/metals.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -153,7 +153,8 @@ lazy val V = new {
   val gradleBloop = bloop
   val mavenBloop = bloop
   val scalafmt = "2.0.1"
-  val mdoc = "2.0.0"
+  val mdoc = "2.0.1+1-dcd42d72-SNAPSHOT"
```
Comment (olafurpg) [inline build.sbt]: Pending new release of mdoc after https://github.com/scalameta/mdoc/pull/208 is merged

**Comment by olafurpg** [comment]
The failing tests seem unrelated 🤔 

```
 X tests.pc.CompletionOverrideConfigSuite.object 79ms 
  tests.TestFailedException: obtained empty completions!
  tests.BaseCompletionSuite$$anonfun$checkEdit$1.apply$mcV$sp(BaseCompletionSuite.scala:83)

...


Nov 17, 2019 5:51:51 AM org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer fireError
SEVERE: An error occurred while processing an incoming message.
java.lang.NullPointerException
	at scala.meta.internal.metals.debug.EndpointLogger.log(EndpointLogger.scala:40)
	at scala.meta.internal.metals.debug.EndpointLogger.consume(EndpointLogger.scala:26)
	at scala.meta.internal.metals.debug.DebugProxy.$anonfun$handleServerMessage$1(DebugProxy.scala:62)
	at scala.meta.internal.metals.debug.EndpointLogger.$anonfun$listen$1(EndpointLogger.scala:33)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at scala.meta.internal.metals.debug.SocketEndpoint.listen(SocketEndpoint.scala:22)
	at scala.meta.internal.metals.debug.EndpointLogger.listen(EndpointLogger.scala:31)
	at scala.meta.internal.metals.debug.DebugProxy.$anonfun$listenToServer$1(DebugProxy.scala:37)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

X tests.DebugProtocolSuite.restart 6670ms 
  java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
```

**Comment by olafurpg** [comment]
The failing test case looks legitimate

```
WARN  thread stop: Evaluating Worksheet Main.worksheet.sc
java.lang.ThreadDeath
	at java.lang.Thread.stop(Thread.java:853)
	at scala.meta.internal.worksheets.WorksheetProvider$$anon$2.$anonfun$run$2(WorksheetProvider.scala:165)
	at scala.meta.internal.metals.Cancelable$$anon$1.cancel(Cancelable.scala:17)
	at scala.meta.internal.metals.Cancelable$.$anonfun$cancelAll$1(Cancelable.scala:28)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.meta.internal.metals.Cancelable$.cancelAll(Cancelable.scala:27)
	at scala.meta.internal.worksheets.WorksheetProvider$$anon$2.run(WorksheetProvider.scala:164)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.lang.NoClassDefFoundError: Could not initialize class scala.reflect.internal.Flags$
```

I have experienced the case where worksheets stop working after killing the thread. Investigating how to fix it...

**Comment by olafurpg** [comment]
The last commit may also fix a bug in the integration with the presentation compiler where a `Thread.stop()` would prevent subsequent completion/hover/signatureHelp request from being evaluated requiring a "reload window" to fix it.

**Comment by tgodzik** [comment]
@olafurpg since this is contained in the other PR, should we merge https://github.com/scalameta/metals/pull/1070  instead?

**Comment by olafurpg** [comment]
Closed in favor of #1070, which includes the commits from this PR

**Pull request closed without merging**